web/packages/design/src/Modal/Modal.tsx

```diff
@@ -16,15 +16,84 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import React from 'react';
-import styled from 'styled-components';
-import PropTypes from 'prop-types';
+import React, { createRef, cloneElement } from 'react';
+import styled, { StyleFunction } from 'styled-components';
+import { createPortal } from 'react-dom';
 
-import { ownerDocument } from './../utils';
-import Portal from './Portal';
-import RootRef from './RootRef';
+type Props = {
+  /**
+   * If `true`, the modal is open.
+   */
+  open: boolean;
+  className?: string;
+  /**
+   * Styles passed to the modal, the parent of the children.
+   */
+  // TODO(ravicious): The type for modalCss might need some work after we migrate the components
+  // that use <Modal> to TypeScript.
+  modalCss?: StyleFunction<any>;
+  children?: React.ReactElement;
```

Comment:
We do want the child to be a single HTML element because that's the assumption that `Modal` makes about its child (it calls `focus` and `setAttribute` on it for example). There's no `React.HTMLNode`, so `React.ReactNode` it's the best I can do here AFAIK.

But I will add comment so it's more clear why `React.ReactElement` is used here.